src/resources/js/components/AccountEdition.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div class="container">
+    <h2>My Account</h2>
+
+    <div v-if="showMessage" class="error">
+      <h3>{{ message }}</h3>
+    </div>
+
+    <form @submit.prevent="updateFirstName()">
+      <label for="updateUsername">
+        <input v-model="newFirstName" type="text" :placeholder="this.User.fname" />
+      </label>
+
+      <input type="submit" value="update firstname" />
+    </form>
+
+    <form @submit.prevent="updateLastName()">
+      <label for="updateUsername">
+        <input v-model="newLastName" type="text" :placeholder="this.User.lname" />
+      </label>
+
+      <input type="submit" value="update lastname" />
+    </form>
+
+    <form @submit.prevent="updateEmail()">
+      <label for="updateEmail">
+        <input v-model="newEmail" type="email" :placeholder="this.User.email" />
+      </label>
+
+      <input type="submit" value="update email" />
+    </form>
+
+    <form @submit.prevent="updatePassword()">
+      <label for="updatePassword">
+        <input v-model="newPassword" type="password" placeholder="New Password" />
+      </label>
+
+      <input type="submit" value="update password" />
+    </form>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+import router from '../routes';
+
+export default {
+  name: 'AccountEdition',
+  props: ['parent'],
+  data() {
+    return {
+      // will use parent data when available (eg. parent.data.user)
+      User: {
+        id: 3,
+        fname: 'Jane',
+        lname: 'Doe',
+        email: 'jane@paul.lo'
+      },
+      newFirstName: '',
```

Comment:
With the single form approach, you can have a User object to contain the changes. `updatedUser`.